code/game/objects/items/miscellaneous.dm

```diff
@@ -12,6 +12,62 @@
 	w_class = WEIGHT_CLASS_SMALL
 	attack_verb = list("warned", "cautioned", "smashed")
 
+/obj/item/herobeacon
+	name = "heroic beacon"
+	desc = "To summon heroes from the past to protect the future."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-blue"
+	item_state = "radio"
+	var/static/list/display_names = list()
+
+/obj/item/herobeacon/attack_self(mob/user)
+	if(user.canUseTopic(src, NO_TK))
+		beacon_hero(user)
+
+/obj/item/herobeacon/proc/beacon_hero(mob/M)
+	if(!display_names.len)
+		var/static/list/herobox = typesof(/obj/item/storage/box/hero)
+		for(var/V in herobox)
+			var/atom/A = V
+			display_names[initial(A.name)] = A
+
+	var/choice = input(M,"What heroic outfit would you like to order?","Historic Heroes") as null|anything in display_names
+	if(!choice || !M.canUseTopic(src, NO_TK) || !in_range(M, src))
```

Comment:
canUseTopic already handles range checks so you can remove in_range, canUseTopic(src, BE_CLOSE, FALSE, NO_TK)

Reply:
I found your review comments at face value not the most helpful without being quite sure what i was looking at, but after some discord discussion i think i've wrapped my head around it. 

Making changes now.